homeassistant/components/asuswrt/router.py

```diff
@@ -57,13 +136,84 @@ async def setup(self) -> None:
         if not self._api.is_connected:
             raise ConfigEntryNotReady
 
+        # Load tracked entities from registry
+        entity_registry = await self.hass.helpers.entity_registry.async_get_registry()
+        track_entities = (
+            self.hass.helpers.entity_registry.async_entries_for_config_entry(
+                entity_registry, self._entry.entry_id
+            )
+        )
+        for entry in track_entities:
+            if entry.domain == TRACKER_DOMAIN:
+                self._devices[entry.unique_id] = AsusWrtDevInfo(
+                    entry.unique_id, entry.original_name
+                )
+
+        # Update devices
+        await self.update_devices()
+
+        self.async_on_close(
+            async_track_time_interval(self.hass, self.update_all, SCAN_INTERVAL)
+        )
+
+    async def update_all(self, now: Optional[datetime] = None) -> None:
+        """Update all AsusWrt platforms."""
+        await self.update_devices()
+
+    async def update_devices(self) -> None:
+        """Update AsusWrt devices tracker."""
+        new_device = False
+        _LOGGER.debug("Checking Devices")
+        try:
+            wrt_devices = await self._api.async_get_connected_devices()
+            if self._connect_error:
```

Comment:
Please only wrap the line that can raise in the try... except block. Use an `else:` block if needed.